src/components/utils/build-select-box-lists.js

```diff
@@ -17,6 +17,8 @@ function createClusterList (clusters, dataCenterId = null) {
       .toList()
       .filter(cluster =>
         cluster.get('canUserUseCluster') &&
+        cluster.get('architecture') &&
```

Comment:
It won't filter clusters with undefined architecture since in that case I think that the cluster is set to "undefined" and not null. But worth checking.

Reply:
If a cluster has it's architecture set to "undefined" in webadmin, it just won't have a `cpu`  data element in its JSON at all.  The transform+reducer will leave `cluster.architecture` undefined.

So...

Undefined in webadmin = missing in JSON = undefined after internal transform = missing in redux = truthy false/

I update the check to force conversion to an actual boolean :
```suggestion
        !!cluster.get('architecture') &&
```

If the value is undefined, 0, false, '', or NaN, the double !! will convert it to `false`.  Any other value will be `true`.